backend/src/main/java/com/mcmasterbaja/live/Serial.java

```
@@ -25,33 +41,45 @@ public static void readLive() {
           // if it does, set the comport to the current port
           comPort = serialPort;
           // break out of the loop
-          setPort = true;
+          exit = true;
           break;
         }
       }
-      if (!setPort) {
-        setPort = true;
+      if (!exit) { // throwing exception if port not found
+        throw new Exception("No suitable port found");
       }
     }
 
-    comPort.setBaudRate(115200);
-    boolean connected = comPort.openPort();
-    System.out.println(connected);
-    // comPort.setComPortTimeouts(SerialPort.TIMEOUT_READ_BLOCKING, 10000, 0);
-    comPort.setComPortTimeouts(SerialPort.TIMEOUT_READ_BLOCKING, 2000, 0);
-
-    FileWriter fw = null;
-    FileWriter fw2 = null;
-    FileWriter fw42 = null;
+    try {
+      comPort.setBaudRate(115200);
+      if (comPort.openPort()) {
+        System.out.println("Connected to port");
+        comPort.setComPortTimeouts(SerialPort.TIMEOUT_READ_BLOCKING, 2000, 0);
+      } else {
+        System.out.println("Could not connect to port.");
+      }
+    } catch (Exception E) {
+      System.out.println("Error during port setup");
+      E.printStackTrace();
```

Should probably throw an error here